### ISSUE TITLE
add JSON-LD versions of schemas

### ISSUE THREAD

**Issue by dwhieb** [digitallinguistics/spec#125]

**Comment by vadekh** [comment]
@dwhieb I was looking into this idea a bit. My takeaway is that we would need to assign each JSON file a `@context` field for it to be a valid JSON-LD file. The context appears to be the first part of the URI address of the schema so for example would the `@context` for `Access` just be the `$id` field i.e. `"@context: http://schemas.digitallinguistics.io/Access-3.1.0.json"`? 
Or would it just be `"@context: http://schemas.digitallinguistics.io/Access-3.1.0"` with no file extension
or `"@context: http://schemas.digitallinguistics.io/"`
or `"@context: http://schemas.digitallinguistics.io/Access-3.1.0.jsonld"`

Basically it seems to me that the `$id` and `$ref` fields would need to either be changed to `@context` or at least redirected to `.jsonld` files.
Also we have to decide how deep we want to dive into the JSON-LD keywords: do we just want the simplest conversion by only assigning the `@context` field? or do we want to use the `@type` and `@id` and `@value` fields etc. to truly take advantage of the JSONLD functionality?

What do you think?

**Comment by rezlinguist** [comment]
==============================
John W. Du Bois
Professor of Linguistics
University of California, Santa Barbara
Santa Barbara, California 93106
USA
dubois@ucsb.edu

**Comment by xrotwang** [comment]
If you do provide JSON-LD versions of your schemas, you might be interested in re-using properties defined in the [CLDF ontology](https://cldf.clld.org/v1.0/terms.rdf).

**Comment by xrotwang** [comment]
More generally (and I admit to being biased here): I think digitallinguistics could profit from integration with [CLDF](https://cldf.clld.org). AFAICT, Daffodil does not provide much in terms of "macro-types" of data, such as wordlists, typological datasets, etc., but rather specifies "micro-types" like "Word". CLDF otoh has specifications for several macro-types, and (at least theoretically - dunno how well things like [Cow](https://csvw-converter.readthedocs.io/en/latest/) work) there's a well-defined conversion from CLDF to JSON-LD. So if digitallinguistics (and Daffodil) operated on top of JSON-LD as converted from CLDF,
- you'd get a specification for macro-types for free
- you'd have a storage format that plays well with version control
- you'd have a bridge to [quite](https://zenodo.org/communities/cldf-datasets/?page=1&size=20) a [bit](https://zenodo.org/communities/dictionaria/?page=1&size=20) of [data](https://zenodo.org/communities/lexibank/?page=1&size=20)

**Comment by xrotwang** [comment]
Btw.: Another - less well specified, but possibly more practical - way to get JSON from CLDF datasets is via having the data served by a Datasette app, as described here: https://github.com/cldf/cookbook/tree/master/recipes/datasette
